pkg/template/helper.go

```diff
@@ -0,0 +1,48 @@
+package template
+
+import "reflect"
+
+// VisitFields visits all fields in the object and call the visitor function on
+// them. In case the object has no fields (like string) then it calls the
+// visitor function on it.
+// The visitor function can manipulate the fields values.
+func VisitFields(obj interface{}, visitor func(string) string) {
+	v := reflect.ValueOf(obj).Elem()
+
+	visit := func(val reflect.Value) {
+		switch val.Kind() {
+		case reflect.String:
+			if val.CanSet() {
+				val.SetString(visitor(val.String()))
+			}
+		case reflect.Ptr, reflect.Interface:
+			if val.Elem().Kind() == reflect.Struct {
+				VisitFields(val.Interface(), visitor)
+			}
+		case reflect.Struct:
```

Comment:
numbers are tricky, can you have parameter reference in json in a number
field without quotes?
On Apr 17, 2015 7:59 PM, "Clayton Coleman" <notifications@github.com> wrote:

> In pkg/template/helper.go
> <https://github.com/openshift/origin/pull/827#discussion_r28615837>:
>
> > +// visitor function on it.
> > +// The visitor function can manipulate the fields values.
> > +func VisitFields(obj interface{}, visitor func(string) string) {
> > +	v := reflect.ValueOf(obj).Elem()
> > +
> > +	visit := func(val reflect.Value) {
> > +		switch val.Kind() {
> > +		case reflect.String:
> > +			if val.CanSet() {
> > +				val.SetString(visitor(val.String()))
> > +			}
> > +		case reflect.Ptr, reflect.Interface:
> > +			if val.Elem().Kind() == reflect.Struct {
> > +				VisitFields(val.Interface(), visitor)
> > +			}
> > +		case reflect.Struct:
>
> Numbers?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/827/files#r28615837>.
>